R/utilities-parameters.R

```diff
@@ -21,40 +21,33 @@ readParametersFromXLS <- function(paramsXLSpath, sheets = NULL) {
     sheets <- c(1)
   }
 
-  paths <- c()
-  values <- c()
-  units <- c()
+  pathsValuesHash <- hash::hash()
+  pathsUnitsHash <- hash::hash()
 
   for (sheet in sheets) {
     data <- readExcel(path = paramsXLSpath, sheet = sheet)
 
     if (!all(columnNames %in% names(data))) {
       stop(messages$errorWrongParamsXLSStructure(paramsXLSpath))
     }
```

Comment:
Using hash for better performance

Reply:
Named vectors will outperform hashes because R's` vector` implementation is in `C`, while `{hash}` package is implemented in `R`.

For example, in this case, hashing is 67 times slower than the vector approach!

``` r
bench::mark(
  "hash" = {
    h <- hash::hash(letters, 1:26)
    h[["a"]]
  },
  "named vector" = {
    v <- 1:26
    names(v) <- letters
    v[["a"]]
  }
)[1:5]
#> # A tibble: 2 x 5
#>   expression        min   median `itr/sec` mem_alloc
#>   <bch:expr>   <bch:tm> <bch:tm>     <dbl> <bch:byt>
#> 1 hash          100.2us  120.6us     7842.    4.21MB
#> 2 named vector    1.4us    1.8us   504813.        0B
```

<sup>Created on 2022-04-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>


Of course, this assumes that creating a named vector is an option. But, if that's not a feasible solution, then maybe hashing is the way to go.